shard-operator-parent/shard-operator-v2/src/main/java/com/redhat/service/smartevents/shard/operator/v2/converters/ManagedBridgeConverter.java

```diff
@@ -0,0 +1,53 @@
+package com.redhat.service.smartevents.shard.operator.v2.converters;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+import com.redhat.service.smartevents.infra.core.api.dto.KafkaConnectionDTO;
+import com.redhat.service.smartevents.infra.core.exceptions.definitions.platform.InvalidURLException;
+import com.redhat.service.smartevents.infra.v2.api.models.dto.BridgeDTO;
+import com.redhat.service.smartevents.shard.operator.v2.resources.DNSConfigurationSpec;
+import com.redhat.service.smartevents.shard.operator.v2.resources.KNativeBrokerConfigurationSpec;
+import com.redhat.service.smartevents.shard.operator.v2.resources.KafkaConfigurationSpec;
+import com.redhat.service.smartevents.shard.operator.v2.resources.ManagedBridge;
+import com.redhat.service.smartevents.shard.operator.v2.resources.TLSSpec;
+
+public class ManagedBridgeConverter {
+
+    public static ManagedBridge fromBridgeDTOToManageBridge(BridgeDTO bridgeDTO, String namespace) {
+        try {
+
+            TLSSpec tlsSpec = new TLSSpec();
+            tlsSpec.setCertificate(bridgeDTO.getTlsCertificate());
+            tlsSpec.setKey(bridgeDTO.getTlsKey());
+
+            DNSConfigurationSpec dns = new DNSConfigurationSpec();
+            dns.setHost(new URL(bridgeDTO.getEndpoint()).getHost());
+            dns.setTls(tlsSpec);
+
+            KafkaConnectionDTO kafkaConnectionDTO = bridgeDTO.getKafkaConnection();
+            KafkaConfigurationSpec kafkaConfigurationSpec = new KafkaConfigurationSpec();
+            kafkaConfigurationSpec.setBootstrapServers(kafkaConnectionDTO.getBootstrapServers());
+            kafkaConfigurationSpec.setPassword(kafkaConnectionDTO.getClientSecret());
+            kafkaConfigurationSpec.setUserId(kafkaConnectionDTO.getClientId());
+            kafkaConfigurationSpec.setSaslMechanism(kafkaConnectionDTO.getSaslMechanism());
+            kafkaConfigurationSpec.setSecurityProtocol(kafkaConnectionDTO.getSecurityProtocol());
+            kafkaConfigurationSpec.setTopicName(kafkaConnectionDTO.getTopic());
+
+            KNativeBrokerConfigurationSpec kNativeBrokerConfigurationSpec = new KNativeBrokerConfigurationSpec();
+            kNativeBrokerConfigurationSpec.setKafkaConfiguration(kafkaConfigurationSpec);
+
+            return new ManagedBridge.Builder()
+                    .withNamespace(namespace)
+                    .withBridgeName(bridgeDTO.getName())
+                    .withCustomerId(bridgeDTO.getCustomerId())
+                    .withOwner(bridgeDTO.getOwner())
+                    .withBridgeId(bridgeDTO.getId())
+                    .withDnsConfigurationSpec(dns)
+                    .withKnativeBrokerConfigurationSpec(kNativeBrokerConfigurationSpec)
```

Comment:
Like it or not; you're going to have to store the `generation` _somewhere_ as it will be needed when an update to the `Conditon`'s is sent back to the Control Plane. The most suitable place is probably in the Custom Resource.

Reply:
will consider it while sending back conditions to manager